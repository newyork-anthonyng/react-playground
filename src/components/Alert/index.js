import React, { Component } from 'react';
import T from 'prop-types';
import glamorous from 'glamorous';

const Container = glamorous.div({
  backgroundColor: '#D3E8F8',
  border: '2px solid #3395E1',
  borderRadius: '4px',
  padding: '10px',
});

const Alert = ({ children }) => (
  <Container
    role="alert"
    aria-live="assertive"
    aria-atomic="true"
  >
    {children}
  </Container>
);

Alert.propTypes = {
  children: T.node.isRequired,
};

export default Alert;
