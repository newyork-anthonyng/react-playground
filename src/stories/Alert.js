import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, number, boolean, text } from '@storybook/addon-knobs';

import Alert from '../components/Alert';

storiesOf('Alert', module)
  .addDecorator(withKnobs)
  .add('Example', () => {
    return (
      <Alert>
        <h1>Alert</h1>
      </Alert>
    );
  })
  .add('With toggle button', () => {
    class Container extends React.Component {
      constructor() {
        super();
        this.state = {
          isOpen: false,
        };

        this.handleButtonClick = this.handleButtonClick.bind(this);
      }

      handleButtonClick() {
        this.setState({ isOpen: !this.state.isOpen });
      }

      render() {
        return (
          <div>
            <button onClick={this.handleButtonClick}>
              {this.state.isOpen ? 'Hide alert' : 'Show alert'}
            </button>
            {
              this.state.isOpen &&
              <Alert>
                <h1>Hey. This is an important message.</h1>
                <h2>You might want to check out your password</h2>
              </Alert>
            }
          </div>
        );
      }
    }

    return <Container />
  })
