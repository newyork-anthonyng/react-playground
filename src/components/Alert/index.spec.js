import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Alert from "./";

it("should render children correctly", () => {
  const wrapper = shallow(
    <Alert>
      <h1>Hello</h1>
      <h2>Yo Yo Yo</h2>
    </Alert>
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});
