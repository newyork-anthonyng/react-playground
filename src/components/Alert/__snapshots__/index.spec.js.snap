// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render children correctly 1`] = `
<glamorous(div)
  aria-atomic="true"
  aria-live="assertive"
  role="alert"
>
  <h1>
    Hello
  </h1>
  <h2>
    Yo Yo Yo
  </h2>
</glamorous(div)>
`;
